src/components/Homepage/HomePage.jsx

```diff
@@ -1,61 +1,105 @@
-      {
-        params: {
-          api_key: api_key,
-        },
-      }
-    );
-    console.log(response);
-    setMovieInfo(response.data);
+  async function fetchData() {
+    setIsLoading(true);
+    try {
+      const response = await axios.get(
+        "https://api.themoviedb.org/3/movie/popular",
+        {
+          params: {
+            api_key: api_key,
+          },
+        }
+      );
+      setTotalCountPages(response.data.total_pages);
+      setMovies(response.data.results);
+      setIsLoading(false);
+    } catch (error) {
+      console.log(error);
+    }
   }
 
-  async function fetchData() {
-    const response = await axios.get(
-      "https://api.themoviedb.org/3/movie/popular",
-      {
-        params: {
-          api_key: api_key,
-        },
-      }
-    );
-    setMovies(response.data.results);
+  async function fetchPageData(page = 1) {
+    setIsLoading(true);
+    try {
+      const response = await axios.get(
+        "https://api.themoviedb.org/3/movie/popular",
+        {
+          params: {
+            api_key: api_key,
+            page: page,
+          },
+        }
+      );
+      setCurrentPage(page);
+      setMovies(response.data.results);
+      setIsLoading(false);
+      navigate(`/page/${page}`);
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  let pagesArray = [];
+  for (let page = 0; page < totalCountPages; page++) {
+    pagesArray.push(page + 1);
   }
 
   return (
     <>
-      <h1 className="text-4xl mt-10">Most popular movies</h1>
-      <div className="mt-20 grid gap-y-10 grid-cols-5">
-        {movies.map((movie) => (
-          <Link
-            to={`movie/${movie.id}`}
-            onClick={() => fetchMovieData(movie.id)}
-            key={movie.id}
-            className="flex items-center flex-col hover:scale-110 transition duration-300"
-          >
-            <img
-              className="rounded"
-              src={`http://image.tmdb.org/t/p/w185/${movie.poster_path}`}
-            />
-            <p className="px-10 py-2">{movie.title}</p>
-          </Link>
-        ))}
-      </div>
+      <h1 className="text-center text-4xl mt-10">Most popular movies</h1>
+      {isLoading ? (
+        <Spinner />
+      ) : (
+        <>
+          <div className="mt-20 grid gap-y-10 grid-cols-5">
+            {movies.map((movie) => (
+              <Link
+                to={`movie/${movie.id}`}
+                key={movie.id}
+                className="flex items-center flex-col hover:scale-110 transition duration-300"
+              >
+                <img
+                  className="rounded"
+                  src={`http://image.tmdb.org/t/p/w185/${movie.poster_path}`}
+                />
+                <p className="px-10 py-2">{movie.title}</p>
+              </Link>
+            ))}
+          </div>
+          <div className="text-center mb-16 mt-4">
+            {pagesArray.slice(0, 10).map((page, index) => (
```

Comment:
There are 32877 pages. You're showing only 10. That's a bad idea. Please think of a better way of implementing the pagination.

Reply:
fixed